compiler/mx.compiler/suite.py

```diff
@@ -955,7 +955,8 @@
       "subDir" : "src",
       "sourceDirs" : ["src"],
       "dependencies" : [
-        "org.graalvm.compiler.replacements.test"
+          "org.graalvm.compiler.replacements.test",
+          "org.graalvm.compiler.hotspot"
```

Comment:
We don't want the vm-independent projects to depend on vm-dependent projects so this dependency is problematic.

One way to solve this is maybe to only keep the "general" test there (`org.graalvm.compiler.replacements.jdk9.test`) and move the aarch64-only test to a hotspot-specific project (e.g., `org.graalvm.compiler.hotspot.jdk9.test`).

Reply:
Ah, well, I thought you were going to say something like that ... :-)

I'm not clear exactly what it means to be a vm-independent vs vm-dependent project (other than the obvious aspect of the package not having hotspot in the name). Does this distinguish between code which runs on/tests only Hotspot vs, say, code which runs on/tests Hotspot and, possibly, other VMs, for example Substrate?

If this is what is at stake then I think I ought to leave some test code which exercises AArch64 in the vm-independent test so that AArch64 code is tested on other VMs i.e. I could leave the part of the code in the AArch64 tests that runs when UseBarriersForVolatile is the default value, false. I could then still add similar tests to a hotspot-specific package that exercise the code for the case when UseBarriersForVolatile is true. Alternatively, I could move all 4 AArch64-specific tests to a hotspot-specific package as they are. What is the best way to proceed?

If I do split the 4 AArch64 tests in to generic (UseBarriersForVolatile=false) and hotspot-specific tests then I'd really need to be able to ensure that the hotspot-specific test is run with UseBarriersForVolatile set to true. Is there a way (other than a command line setting) to force that config option to be set to a specific value before the test is run?

Reply:
> I'm not clear exactly what it means to be a vm-independent vs vm-dependent project (other than the obvious aspect of the package not having hotspot in the name). Does this distinguish between code which runs on/tests only Hotspot vs, say, code which runs on/tests Hotspot and, possibly, other VMs, for example Substrate?

Yes, by vm-independent i mean things that will be sued/run on other VMs than HotSpot such as SubstrateVM.

The important point i think is that the vm-independent version should ideally pass regardless of the VM and its flags. This means that the generic test can probably not really be tailored for `UseBarriersForVolatile=false` since setting that to true would break the test.

This is why i would always `assumeFalse(arch instanceof AArch64)` in the generic tests and add aarch64-specific tests (`assumeTrue(arch instanceof AArch64)`) in a hotspot related package where we can check the flag status and decide which version to expect.

Reply:
Regarding setting VM options from a test: you can not unless you launch a different VM explicitly. We don't usually do that in tests and rather rely on running the unit tests in differnt configurations externally. If you want to launch an other VM, you can see for example `CompilationWrapperTest` (which again, is hotspot-specific)

Reply:
Hi Gilles,

Thanks for the quick response.

Ok, I'll patch the existing tests to remove the hotspot dependency and the special handling of UseBarriersForVolatile.

I'll also look into adding a Hotspot-specific test following the model of `CompilationWrapperTest` to exercise tests with UseBarriersForVolatile=true.

I'll post a follow-up comment when these are ready.